codersdk/workspaces.go

```diff
@@ -218,9 +218,14 @@ func (c *Client) PutExtendWorkspace(ctx context.Context, id uuid.UUID, req PutEx
 
 type WorkspaceFilter struct {
 	OrganizationID uuid.UUID `json:"organization_id,omitempty"`
-	// Owner can be a user_id (uuid), "me", or a username
+	// Owner can be "me" or a username
 	Owner string `json:"owner,omitempty"`
-	Name  string `json:"name,omitempty"`
+	// Template is a template name
+	Template string `json:"template,omitempty"`
+	// Name will return partial matches
+	Name string `json:"name,omitempty"`
+	// FilterQuery supports a raw filter query string
+	FilterQuery string `json:"q,omitempty"`
```

Comment:
This seems pretty non-standard to support both. When would this be used?

Reply:
I was on the fence with this as well. Github supports both a full query and an array of tags to list issues on their api for example. I'm in favor of removing the query params and add them back later if we need to. @Emyrk thoughts?

Reply:
@kylecarbs from a curl + cli perspective I figured it's easier to use queryparams.

From the UI, it's easier to build a single string. :shrug:.

Reply:
@kylecarbs It is better to parse the query param on the backend imo, since the backend controls the actual behavior of the search query. Makes the FE job easier.

But from an api perspective, I really just figured query params are easier. We use query params in other places, so if we only support `q` and not the others, we have different behavior depending on the endpoint.

Reply:
I have to convert the query to params regardless. So it's not really any more code.

Reply:
It seems non-standard to have both from my perspective. What do you think about disallowing other filters in favor of the queryable format?